File: common-util/src/main/java/org/study/common/util/utils/AESUtil.java
package org.study.common.util.utils;

import org.study.common.statics.exceptions.BizException;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;

/**
 * AES加解密工具类
 * @author chenyf
 * @date 2018-12-15
 */
public class AESUtil {
    public final static String ALG_AES = "AES";
    public static final String ENCODING_UTF_8 = "UTF-8";
    private static final String ECB_MODE = "AES/ECB/PKCS5Padding";//算法/模式/补码方式

    public static String encryptECB(String content, String secKey){
        try {
            Cipher cipher = Cipher.getInstance(ECB_MODE);
            SecretKeySpec secSpec = genSecretKeySpec(secKey);
            cipher.init(Cipher.ENCRYPT_MODE, secSpec);
            byte[] encrypted = cipher.doFinal(content.getBytes(ENCODING_UTF_8));
            return CodeUtil.base64Encode(encrypted);
        }catch(Throwable e){
            throw new BizException("AES加密异常", e);
        }
    }

    public static String decryptECB(String content, String secKey){
        try {
            Cipher cipher = Cipher.getInstance(ECB_MODE);
            SecretKeySpec secSpec = genSecretKeySpec(secKey);
            cipher.init(Cipher.DECRYPT_MODE, secSpec);
            byte[] encrypted1 = CodeUtil.base64Decode(content);
            byte[] original = cipher.doFinal(encrypted1);
            return new String(original, ENCODING_UTF_8);
        }catch(Throwable e){
            throw new BizException("AES解密异常", e);
        }
    }

    /**
     * 生成密钥对象 密钥可支持16位或32位，如果是32位，可能会报：java.security.InvalidKeyException: Illegal key size 异常，此时需要更换JDK的local_policy.jar和US_export_policy.jar
     * @param secKey
     * @return
     * @throws Exception
     */
    public static SecretKeySpec genSecretKeySpec(String secKey) throws Exception {
        if (secKey == null || (secKey.length() != 16 && secKey.length() != 32)) {
            throw new BizException("密钥长度须为16或32位");
        }
        return new SecretKeySpec(secKey.getBytes(ENCODING_UTF_8), ALG_AES);
    }
}

File: demo-gateway-phpclient/utils/AESUtil.php
<?php


namespace utils;


use exceptions\BizException;

/**
 * AES加解密工具类
 * Class AESUtil
 * @package utils
 */
class AESUtil {
    const EBC_MODE = "AES-128-ECB";
//    const CBC_MODE = "AES-128-CBC";

    /**
     * AES加密，模式为：AES/ECB/PKCK7Padding
     * @param string $data
     * @param string $secKey
     * @return string
     * @throws BizException
     */
    public static function encryptECB(string $data, string $secKey){
        $encrypted = openssl_encrypt($data, self::EBC_MODE, $secKey, OPENSSL_RAW_DATA);
        if($encrypted === false){
            throw new BizException(BizException::BIZ_ERROR, "aes加密失败");
        }
        return base64_encode($encrypted);
    }

    /**
     * AES解密，模式为：AES/ECB/PKCK7Padding
     * @param string $data
     * @param string $secKey
     * @return string
     * @throws BizException
     */
    public static function decryptECB(string $data, string $secKey){
        $decrypted = openssl_decrypt(base64_decode($data), self::EBC_MODE, $secKey, OPENSSL_RAW_DATA);
        if($decrypted === false){
            throw new BizException(BizException::BIZ_ERROR, "aes解密失败");
        }
        return $decrypted;
    }

//    public static function encryptCBC(string $data, string $secKey, string $iv){
//        $encrypted = openssl_encrypt($data, self::CBC_MODE, $secKey, OPENSSL_RAW_DATA, $iv);
//        if($encrypted === false){
//            throw new BizException(BizException::BIZ_ERROR, "aes加密失败");
//        }
//        return base64_encode($encrypted);
//    }

//    public static function decryptCBC(string $data, string $secKey, string $iv){
//        $decrypted = openssl_decrypt(base64_decode($data), self::CBC_MODE, $secKey, OPENSSL_RAW_DATA, $iv);
//        if($decrypted === false){
//            throw new BizException(BizException::BIZ_ERROR, "aes解密失败");
//        }
//        return self::pkcs5_unpad($decrypted);
//    }
}